tests/kafkatest/tests/client/run_compatibility_test.py

```diff
@@ -0,0 +1,92 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.mark import parametrize
+from ducktape.tests.test import TestContext
+
+from kafkatest.directory_layout.kafka_path import KafkaPathResolverMixin
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.kafka import KafkaService
+from ducktape.tests.test import Test
+from kafkatest.version import TRUNK, V_0_10_0_1, KafkaVersion
+
+def get_broker_features(broker_version):
+    features = {}
+    if (broker_version < V_0_10_0_1):
+        features["offsetsForTimesSupported"] = False
+        features["clusterIdSupported"] = False
+    else:
+        features["offsetsForTimesSupported"] = True
+        features["clusterIdSupported"] = True
+    return features
+
+def run_command(node, cmd, ssh_log_file):
+    with open(ssh_log_file, 'w') as f:
+        f.write("Running %s\n" % cmd)
+        try:
+            for line in node.account.ssh_capture(cmd):
+                f.write(line)
+        except Exception as e:
+            f.write("** Command failed!")
+            print e
+            raise e
+
+
+class RunCompatibilityTest(Test):
+    """
+    Test running the CompatibilityTest.  It will test for the presence or
+    absence of specific features.
+    """
+
+    def __init__(self, test_context):
+        """:type test_context: ducktape.tests.test.TestContext"""
+        super(RunCompatibilityTest, self).__init__(test_context=test_context)
+
+        self.zk = ZookeeperService(test_context, num_nodes=3)
+        self.topics = { "test_topic": {
+            "partitions": 10,
+            "replication-factor": 1
+            }}
+        self.kafka = KafkaService(test_context, num_nodes=3, zk=self.zk, topics=self.topics)
+
+    def invoke_compatibility_program(self, features):
+        # Run the compatibility test on the first Kafka node.
+        node = self.zk.nodes[0]
+        cmd = ("%s org.apache.kafka.tools.CompatibilityTest "
+               "--bootstrap-servers %s "
+               "--offsetsForTimesSupported %s "
+               "--clusterIdSupported %s "
+               "--topic %s " % (self.zk.path.script("kafka-run-class.sh", node),
+                               self.kafka.bootstrap_servers(),
+                               features["offsetsForTimesSupported"],
+                               features["clusterIdSupported"],
+                               self.topics.keys()[0]))
+        ssh_log_file = "%s/%s" % (TestContext.results_dir(self.test_context, 0),
+                                  "compatibility_test_output.txt")
+        try:
+          self.logger.info("Running %s" % cmd)
+          run_command(node, cmd, ssh_log_file)
+        except Exception as e:
+          self.logger.info("** Command failed.  See %s for log messages." % ssh_log_file)
+          raise e
+
+    # TODO: when KAFKA-4462 is complete, we should test other supported versions here.
```

Comment:
Yeah, we should be testing older versions here now.  I'll update test_producer_consumer_compat.py as well.